Week3/homework/step2-6.js

```diff
@@ -2,20 +2,27 @@
 
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
+const arrXd = [[[[1]], [2]], [[[[[3]]]], 4], [[5], [[6]], [7]], [[[[[[8, 9]]]]]]];
+
+const flattenAll = arr =>
+  arr.reduce((resultArr, item) => {
+    if (Array.isArray(item)) {
+      return [...resultArr, ...flattenAll(item)]; // Recursion
+    }
+    return [...resultArr, item];
+  }, []);
```

Comment:
I'm going to save this piece of logic for future 👏

Reply:
Thanks indeed for your supportive comment  🙏